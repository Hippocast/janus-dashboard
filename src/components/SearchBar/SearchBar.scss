@import 'index.scss';

.j-search-bar {
    position: relative;
    width: 360px;
    background-color: $hf-color-white;

    &__input {
        position: relative;
        width: 100%;
        height: 36px;
        padding-left: 50px;
        background-color: transparent;
    }

    &__icon {
        position: absolute;
        width: 24px;
        height: 24px;
        left: 16px;
        top: 50%;
        transform: translateY(-50%);
        background-image: url(../../assets/img/grey.svg);
    }
}

.j-row__item:not(:last-child) > .j-search-bar {
    margin-right: $hf-padding-middle;
}
