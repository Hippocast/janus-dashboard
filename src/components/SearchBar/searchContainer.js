import { connect } from 'react-redux'

import {
  setSearchQuery,
  discardPagination
} from '../../store/actions'

import SearchBar from './SearchBar'

const mapStateToProps = state => ({
  searchQuery: state.searchReducer.searchQuery
})

export default connect(
  mapStateToProps,
  { setSearchQuery, discardPagination }
)(SearchBar)
