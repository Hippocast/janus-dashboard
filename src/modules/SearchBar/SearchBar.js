import React from 'react';
import PropTypes from 'prop-types';

import block from '../../helpers/bem-cn';

import '../forms/FormInput/FormInput.css';
import './SearchBar.css';

const propTypes = {
  discardPagination: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  searchQuery: PropTypes.string,
  setSearchQuery: PropTypes.func.isRequired,
};

const defaultProps = {
  value: '',
  placeholder: 'Search...',
};

const b = block('j-search-bar');

const SearchBar = ({ discardPagination, placeholder, searchQuery, setSearchQuery }) => {
  return (
    <div className={b}>
      <span className={b('icon')}></span>
      <input
        className={b('input').mix('j-input')}
        type="text"
        placeholder={placeholder}
        value={searchQuery}
        onChange={({ target: { value } }) => {
          discardPagination();
          setSearchQuery(value);
        }}
      />
    </div>
  );
};

SearchBar.propTypes = propTypes;
SearchBar.defaultProps = defaultProps;

export default SearchBar;
