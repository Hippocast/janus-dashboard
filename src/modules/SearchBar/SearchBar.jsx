import React from 'react';
import PropTypes from 'prop-types';

import SearchIcon from '../UI/icons/SearchIcon';

import '../forms/FormInput/FormInput.css';
import './SearchBar.css';

const propTypes = {
  discardPagination: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  searchQuery: PropTypes.string,
  setSearchQuery: PropTypes.func.isRequired,
};

const defaultProps = {
  value: '',
  placeholder: 'Search...',
};

const SearchBar = ({ discardPagination, placeholder, searchQuery, setSearchQuery }) => {
  return (
    <div className="j-search-bar">
      <SearchIcon className="j-search-bar__icon" />
      <input
        className="j-input j-search-bar__input"
        type="search"
        placeholder={placeholder}
        value={searchQuery}
        onChange={({ target: { value } }) => {
          discardPagination();
          setSearchQuery(value);
        }}
      />
    </div>
  );
};

SearchBar.propTypes = propTypes;
SearchBar.defaultProps = defaultProps;

export default SearchBar;
