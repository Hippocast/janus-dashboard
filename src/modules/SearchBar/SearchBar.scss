@import 'index.scss';

.j-search-bar {
  position: relative;
  width: 360px;
  background-color: $hf-color-white;

  &__input {
    position: relative;
    width: 100%;
    padding-left: 50px;
    background-color: transparent;
  }

  &__icon {
    position: absolute;
    width: 24px;
    height: 24px;
    left: 16px;
    top: 50%;
    transform: translateY(-50%);
  }
}
