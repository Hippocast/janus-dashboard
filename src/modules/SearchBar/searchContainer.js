import { connect } from 'react-redux';

import {
  setSearchQuery,
  discardPagination,
} from '../../store/actions';

import SearchBar from './SearchBar';

const mapStateToProps = state => {
  return {
    searchQuery: state.searchReducer.searchQuery,
  };
};

export default connect(
  mapStateToProps,
  { setSearchQuery, discardPagination },
)(SearchBar);
