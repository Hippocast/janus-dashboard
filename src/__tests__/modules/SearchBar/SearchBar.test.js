import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '../../../modules/SearchBar/SearchBar';

describe('SearchBar component', () => {
    const passedProps = {
        discardPagination: jest.fn(),
        placeholder: 'Search...',
        searchQuery: '',
        setSearchQuery: jest.fn(),
    };

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <SearchBar
                    {...passedProps}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('handles onChange', () => {
        const otherProps = {
            placeholder: 'Search...',
            searchQuery: '',
        };

        it('triggers discarding pagination', () => {
            const discardPagination = jest.fn();
            const component = renderer.create(
                <SearchBar
                    discardPagination={discardPagination}
                    setSearchQuery={passedProps.setSearchQuery}
                    {...otherProps}
                />
            );

            let tree = component.toJSON();
            tree.children[1].props.onChange({ target: { value: 'some'}});

            expect(discardPagination).toBeCalled();

            tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('triggers setting new search query', () => {
            const setSearchQuery = jest.fn()
                .mockReturnValue('default');
            const component = renderer.create(
                <SearchBar
                    discardPagination={passedProps.discardPagination}
                    setSearchQuery={setSearchQuery}
                    {...otherProps}
                />
            );

            let tree = component.toJSON();
            tree.children[1].props.onChange({ target: { value: 'some'}});

            expect(setSearchQuery).toBeCalled();

            tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
