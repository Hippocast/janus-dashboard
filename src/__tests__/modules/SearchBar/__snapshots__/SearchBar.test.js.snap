// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component handles onChange triggers discarding pagination 1`] = `
<div
  className={[Function]}
>
  <span
    className={[Function]}
  />
  <input
    className={[Function]}
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</div>
`;

exports[`SearchBar component handles onChange triggers setting new search query 1`] = `
<div
  className={[Function]}
>
  <span
    className={[Function]}
  />
  <input
    className={[Function]}
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</div>
`;

exports[`SearchBar component renders correctly 1`] = `
<div
  className={[Function]}
>
  <span
    className={[Function]}
  />
  <input
    className={[Function]}
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</div>
`;
